src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -0,0 +1,134 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "./BLSRegistryCoordinatorWithIndices.sol";
+
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IBLSRegistryCoordinatorWithIndices.sol";
+
+/**
+ * @title BLSOperatorStateRetriever with view functions that allow to retrieve the state of an AVSs registry system.
+ * @author Layr Labs Inc.
+ */
+contract BLSOperatorStateRetriever {
+    struct Operator {
+        bytes32 operatorId;
+        uint96 stake;
+    }
+
+    struct CheckSignaturesIndices {
+        uint32[] nonSignerQuorumBitmapIndices;
+        uint32[] quorumApkIndices;
+        uint32[] totalStakeIndices;  
+        uint32[][] nonSignerStakeIndices; // nonSignerStakeIndices[quorumNumberIndex][nonSignerIndex]
+    }
+
+    /**
+     * @notice This function is intended to by AVS nodes every time there is new triggered tasks. Since all of 
```

Comment:
`to by` => `to be called by`?
`there is new triggered tasks` => `a new task is created for the AVS`?
not sure what the intended meaning of this sentence is, exactly

also `of crucial` => `of the crucial`

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/be2efe90a2630d953b4da4757b4dd40427df688f does this help?

Reply:
yes